### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by mvz**
## Summary

Various code cleanups

## Details

- Remove :have_option_value matcher
- Remove :be_valid_option matcher
- Simplify set_environment_variable features
- Temporarily disable RSpec/PredicateMatcher cop
- Correct Gemspec/RequiredRubyVersion offenses
- Disable AmbiguousX cops
- Fix Lint/MissingSuper offenses
- Replace custom include_an_object matcher with RSpec's include matcher
- Use RSpec's all matcher the regular way
- Remove obsolete requires of 'rspec/expectations/version'
- Remove unused ObjectFormatter and BaseMatcher classes
- Move event bus specs to their logical location
- Clean up event bus specs
- Fix Lint/Void offenses
- Fix Naming/ConstantName offense
- Improve scenario identifiers when reporting scenario timing
- Regenerate RuboCop to-do file

## Motivation and Context

The new RuboCop release was erroring on our spec support code loaders, and I took a look at those support files. This lead to removing some of the support logic that wasn't really earning its keep. After that, I just felt like fixing some of the items in the RuboCop to-do file.

## How Has This Been Tested?

I ran the specs and cukes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Internal change (refactoring, test improvements, developer experience or update of dependencies)

## Checklist:

- [ ] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: features/04_aruba_api/environment/set_environment_variable.feature [hunk 1]
```diff
@@ -12,58 +12,71 @@ Feature: Set environment variable via API-method
...
+        expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=1'
+        expect(ENV['LONG_LONG_VARIABLE']).to be_nil
+      end
     end
     """
     When I run `rspec`
     Then the specs should all pass
 
-  Scenario: Existing variable set from within the test
+  Scenario: Overriding an previously set variable
```
Comment (olleolleolle) [inline features/04_aruba_api/environment/set_environment_variable.feature:37]:
```suggestion
  Scenario: Overriding a previously set variable
```

File: features/04_aruba_api/environment/set_environment_variable.feature [hunk 2]
```diff
@@ -12,58 +12,71 @@ Feature: Set environment variable via API-method
...
+        expect(last_command_started.output).to include 'LONG_LONG_VARIABLE=2'
+      end
     end
     """
     When I run `rspec`
     Then the specs should all pass
 
 
-  Scenario: Existing variable set by some outer parent process
+  Scenario: Setting variable set by some outer parent process
```
Comment (olleolleolle) [inline features/04_aruba_api/environment/set_environment_variable.feature:60]: Does this read right, now? Can it be made shorter and still mean the right thing?

File: features/04_aruba_api/environment/set_environment_variable.feature [hunk 3]
```diff
@@ -97,43 +115,43 @@ Feature: Set environment variable via API-method
...
     end
     """
     When I run `rspec`
     Then the specs should all pass
 
   Scenario: Run some ruby code in code with previously set environment
 
-    The `#with_environment`-block makes the change environment temporary
+    The `#with_environment`-block makes the changed environment temporarily
     avaiable for the code run within the block.
```
Comment (olleolleolle) [inline features/04_aruba_api/environment/set_environment_variable.feature:137]:
```suggestion
    available for the code run within the block.
```

**Review by olleolleolle — APPROVED** [review]
I read through it, and it looks 👍

File: features/04_aruba_api/environment/set_environment_variable.feature (hunk 2 above)
Comment (mvz) [reply features/04_aruba_api/environment/set_environment_variable.feature]: I simplified the description.

**Pull request merged**